molecules_app/main.py

```diff
@@ -12,47 +11,28 @@ class Molecule(BaseModel):
     weight: str
 
 
-file_path = "C:\\Users\\gabi0\\Desktop\\chemical.txt"
-
-
 @app.post("/molecules")
-async def add_molecule(molecule: Molecule | list[Molecule]):
-    p = Path(file_path)
-    if not p.exists():
-        with p.open('w') as file:
-            content = json.dumps([])
-            file.write(content)
+async def add_molecule(molecule: Molecule | list[Molecule], file_path):
     with open(file_path, "r") as file:
         content = json.load(file)
-        content.append(molecule.model_dump())
+        if type(molecule) is list:
```

Comment:
Mogłabyś tu użyć funkcji `isinstance` zamiast tego? Tej funkcji się używa w produkcyjnym kodzie Pythonowym do sprawdzania typów. Ona jest trochę bardziej "mądra". Jak chcesz to tu możesz sobie poczytać o różnicach: https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance

Ją używasz tak `isinstance(molecule, list)`